app/components/Activity/ActivityActions/ActivitySearch.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import debounce from 'lodash.debounce'
+import { Form, Input } from 'components/UI'
+
+const ActivitySearch = ({ searchQuery, placeholder, updateActivitySearchQuery, ...rest }) => {
+  const debouncedUpdateActivitySearchQuery = debounce(updateActivitySearchQuery, 300)
```

Comment:
In theory this will create debounced function on each render causing debounce to work incorrectly. I think it's better to create one with `useEffect` hook

Reply:
I took a stab at implementing this using hooks, but it wasn't really working as I wanted it to. Rather than spend too much time on this here I have converted this to a `PureComponent` and set up the debounce as a class method to remove the redundant renders. This the same approach as we take elsewhere in the app.

Converting all of our debounce to a standardised hook implementation would make a nice future refactor PR.

Reply:
I wonder what was the issue you were having and/or the particular hook you were using.

Reply:
Got this working and updated the code top use new `useDebounce` hook